### PR TITLE
[FIXES-#2228,TEST] handle empty input correctly in translate()

### LINKED ISSUE #2228: translation module crashes when fed an empty sequence
see seqan/lambda#111

### PR DESCRIPTION
**Pull request by h-2**
add sanity checks and tests that provoke them.

fixes #2228 / seqan/lambda#111

### CODE REVIEW

**Comment by h-2** [comment]
!rebuild

File: include/seqan/translation/translation.h [hunk 1]
```diff
@@ -312,8 +312,8 @@ _translateImplLoopOMPWrapper(TTarget & target,
 {
     SEQAN_OMP_PRAGMA(parallel for schedule(dynamic))
     for (int64_t i = 0; i < static_cast<int64_t>(length(target)); ++i)
-        _translateImplLoop(target, i, source, GeneticCode<CODE_SPEC>(),
-                           Frames_<frames>());
+        if (length(source[i/frames]) >= 3)
```
Comment (rrahn) [inline include/seqan/translation/translation.h]: Please add braces around the for-loop body.

File: include/seqan/translation/translation.h [hunk 2]
```diff
@@ -327,8 +327,8 @@ _translateImplLoopOMPWrapper(TTarget & target,
 {
     typedef typename Size<TTarget>::Type TPos;
     for (TPos i = 0; i < length(target); ++i)
-        _translateImplLoop(target, i, source, GeneticCode<CODE_SPEC>(),
-                           Frames_<frames>());
+        if (length(source[i/frames]) >= 3)
```
Comment (rrahn) [inline include/seqan/translation/translation.h]: Same here.
Could you also comment the if-condition. As it is not a complete trivial condition here.

File: tests/translation/test_translation.h [hunk 3]
```diff
@@ -596,6 +596,42 @@ test_translation_stringset_multiframe_impl0()
         appendValue(source, str);
```
Comment (rrahn) [inline tests/translation/test_translation.h]: there is double new-line. Was there already, but you can as well just remove it here.

Comment (h-2) [reply tests/translation/test_translation.h]: hm, separate white space changes from functional ones, though, or not?

File: include/seqan/translation/translation.h (hunk 1 above)
Comment (h-2) [reply include/seqan/translation/translation.h]: it's all one line statements, that doesn't violate our rules :point_up:

File: include/seqan/translation/translation.h (hunk 2 above)
Comment (h-2) [reply include/seqan/translation/translation.h]: done

File: include/seqan/translation/translation.h (hunk 1 above)
Comment (rrahn) [reply include/seqan/translation/translation.h]: 😒

**Comment by h-2** [comment]
please merge when travis is through

**Pull request merged**